### PR TITLE
[SOAR-3212] - Recorded Future Update Request Headers

### PR DESCRIPTION
**Pull request by jmcadams-r7**
## Proposed Changes:

This PR adds "insightconnect/{plugin_version}" to all outgoing API requests. (This will make us compliant with RF's guidelines on official integrations)

## Validation: 

```
rmt-mbp-5357:recorded_future jmcadams$ icon-validate .
[*] Validating plugin at .

[*] Running Integration Validators...
[*] Executing validator HelpValidator
[*] Executing validator ChangelogValidator
[*] Executing validator RequiredKeysValidator
[*] Executing validator UseCaseValidator
[*] Executing validator SpecPropertiesValidator
[*] Executing validator SpecVersionValidator
[*] Executing validator FilesValidator
[*] Executing validator TagValidator
[*] Executing validator DescriptionValidator
[*] Executing validator TitleValidator
[*] Executing validator VendorValidator
[*] Executing validator DefaultValueValidator
[*] Executing validator IconValidator
[*] Executing validator RequiredValidator
[*] Executing validator VersionValidator
[*] Executing validator DockerfileParentValidator
[*] Executing validator LoggingValidator
[*] Executing validator ProfanityValidator
[*] Executing validator AcronymValidator
[*] Executing validator JSONValidator
[*] Executing validator OutputValidator
[*] Executing validator RegenerationValidator
[*] Executing validator HelpInputOutputValidator
[*] Executing validator SupportValidator
[*] Executing validator RuntimeValidator
[*] Plugin successfully validated!

----
[*] Total time elapsed: 1406.091ms
```

## Testing: 

`./run.sh -R all`

Results ommited due to size. Tests ran as expected.

### CODE REVIEW

**Review by mhofert-r7 — DISMISSED** [review]

File: recorded_future/komand_recorded_future/connection/connection.py [hunk 1]
```diff
@@ -14,7 +14,22 @@ def __init__(self):
...
+        self.client = ApiV2Client(auth=self.token, app_name=self.app_name, app_version=self.app_version)
 
     def test(self):
         return demo_test.demo_test(self.token, self.logger)
+
+    def setup_custom_header(self):
+        with open("../../plugin.spec.yaml") as f:
+            spec = yaml.load(f)
+
+        return spec.get("version")
```
Comment (mrinehart-r7) [inline recorded_future/komand_recorded_future/connection/connection.py:33]: You should be able to get this information from the runtime directly

Comment (mrinehart-r7) [reply recorded_future/komand_recorded_future/connection/connection.py:33]: `self.connection.meta.version`

**Review by mrinehart-r7 — CHANGES_REQUESTED** [review]

File: recorded_future/komand_recorded_future/connection/connection.py (hunk 1 above)
Comment (jmcadams-r7) [reply recorded_future/komand_recorded_future/connection/connection.py:33]: I looked for that via the debug inspector and it wasn't available.

Comment (jmcadams-r7) [reply recorded_future/komand_recorded_future/connection/connection.py]: Fixed (sort of)

File: recorded_future/komand_recorded_future/connection/connection.py [hunk 2]
```diff
@@ -4,6 +4,7 @@
 from rfapi import ApiV2Client
 from komand_recorded_future.util import demo_test
 import yaml
+import pprint
```
Comment (mrinehart-r7) [inline recorded_future/komand_recorded_future/connection/connection.py:7]: Unused import

**Review by mrinehart-r7 — CHANGES_REQUESTED** [review]

File: recorded_future/komand_recorded_future/connection/connection.py [hunk 3]
```diff
@@ -14,7 +15,27 @@ def __init__(self):
...
 
     def test(self):
         return demo_test.demo_test(self.token, self.logger)
+
+    def setup_custom_header(self):
+        version = "test-version"
+        try: # This may not be defined in local komand instances.
+            version = self.meta.version
+        except:
+            pass
```
Comment (mrinehart-r7) [inline recorded_future/komand_recorded_future/connection/connection.py:33]:
```suggestion
            version = "test-version"
```

**Review by mrinehart-r7 — CHANGES_REQUESTED** [review]

**Review by mrinehart-r7 — APPROVED** [review]

**Review by ncramer-r7 — APPROVED** [review]

**Pull request merged**